website/pages/docs/theming/advanced.mdx

```diff
@@ -97,3 +97,44 @@ interface StyleConfig {
   }
 }
 ```
+
+## Theme Typings
+
+<Badge fontSize="sm" colorScheme="teal" letterSpacing="wider">
+  v1.2.0
+</Badge>
+
+When adding new components, component variations, sizes, colors and other theme
+foundations, you can extend the internal theme typings to provide autocomplete
+for your application theme.
+
+If you want to learn how to scale your custom theme you can
+[follow this guide](/docs/theming/customize-theme#scaling-out-your-project).
+
+```bash
+npx @chakra-ui/cli tokens <path/to/your/theme.(js|ts)>
+```
+
+The theme entrypoint file should export the theme object either as `default`
+export or as named `theme` export.
+
+> Note 🚨: If you delete the `node_modules` directory, you'll need to re-run the
+> command to get proper typings again.
+
+For convenience, you can add a `postinstall` script to your `package.json`, so you
+don't have to think about this every time you reinstall your dependencies.
+
+```json
+"scripts": {
+  "gen:theme-typings": "chakra-cli tokens <path/to/your/theme.(js|ts)>",
+  "postinstall": "npm run gen:theme-typings"
+}
+```
+
+The autocomplete for theme token **only works in _Visual Studio Code_ for now**.
+We keep an eye on the TypeScript support of Jetbrains IDEs, but I we have to
+wait until it is improved.
+
+Please note that the CLI cannot read non-JavaScript/TypeScript files. If you
+import other file types (`.css`, `.woff` or `.svg`) in your theme file, you'll
+need to move those imports out of the theme file.
```

Comment:
I feel like these both probably deserve to live in an `Alert` or `>` blockquote to call more attention to them, just to make it a little less likely that we have issues opened for them.